pkgplus/project/service.go

```diff
@@ -398,12 +400,38 @@ func (s *Service) RunContainer(stop <-chan bool, updates chan<- ServiceRunUpdate
 				Status:      ServiceRunStatus_Error,
 			}
 			return err
-		case <-okChan:
-			updates <- ServiceRunUpdate{
-				ServiceName: s.Name,
-				Message:     "Service successfully exited",
-				Status:      ServiceRunStatus_Done,
+		case okBody := <-okChan:
+			if okBody.StatusCode != 0 {
+				logOptions := types.ContainerLogsOptions{ShowStdout: true, ShowStderr: true, Tail: "20"}
+				logReader, err := dockerClient.ContainerLogs(context.Background(), containerId, logOptions)
+
+				if err != nil {
+					return err
+				}
+
+				// Create a buffer to hold the logs
+				var logs bytes.Buffer
+				if _, err := stdcopy.StdCopy(&logs, &logs, logReader); err != nil {
+					return fmt.Errorf("error reading logs for service %s: %w", s.Name, err)
+				}
+
+				err = fmt.Errorf("service %s exited with non 0 status\n %s", s.Name, logs.String())
+
+				updates <- ServiceRunUpdate{
+					ServiceName: s.Name,
+					// TODO: Extract the error logs for the container here...
+					Err:    err,
+					Status: ServiceRunStatus_Error,
+				}
+				return err
```

Comment:
If we want to handle this in a tea program instead, just need to make this return nil and have the tea program handle the updates channel instead.